filebeat/input/filestream/defaultconfig_windows.go

```diff
@@ -0,0 +1,37 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build windows
+
+package filestream
+
+import "time"
+
+func defaultCloserConfig() closerConfig {
+	return closerConfig{
+		OnStateChange: stateChangeCloserConfig{
+			CheckInterval: 5 * time.Second,
+			Removed:       true, // It's ok to leave clean_removed: true
+			Inactive:      5 * time.Minute,
+			Renamed:       false,
+		},
+		Reader: readerCloserConfig{
+			OnEOF:         false,
+			AfterInterval: 0 * time.Second,
+		},
+	}
```

Comment:
Let's extract this into another function, so we can use it here and just modify a single default value instead of redefining all defaults again. We could call it `defaultCloserConfigCommon` and then `defaultCloserConfig` would just be:

```go
return defaultCloserConfigCommon()
```

and on Windows

```go
c := defaultCloserConfigCommon()
c.OnStateChange.Removed = true
return c
```

It avoids duplication and makes it more obvious what the difference is.

Otherwise, it's likely we'll miss this if we want to change more defaults in the future.

Reply:
It makes sense, thanks.